addons/sale_pdf_quote_builder/models/sale_order.py

```diff
@@ -8,6 +8,13 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    def _default_quotation_document_ids(self):
+        return self.env['quotation.document'].search([
+            *self.env['quotation.document']._check_company_domain(self.env.companies),
```

Comment:
```suggestion
            *self.env['quotation.document']._check_company_domain(self.env.company),
```
Since the default `company_id` of the SO is `self.env.company`, we should only add the default documents for the main current company and not all the current companies enabled in the company switcher.